keytableprinter/keytableprinter.go

```diff
@@ -2,94 +2,179 @@ package keytableprinter
 
 import (
 	"fmt"
+	"sort"
 	"strings"
 
 	"github.com/fluidkeys/fluidkeys/colour"
 	"github.com/fluidkeys/fluidkeys/pgpkey"
 )
 
+type column = []string
+type row = []string
+
 const gutter = "  "
+const divider = "---"
 
-var headers = [3]string{
-	"Email address",
-	"Created",
-	"Next rotation",
+var header = row{
+	colour.Info("Email address"),
+	colour.Info("Created"),
+	colour.Info("Status"),
 }
 
+var placeholderDividerRow = row{divider, divider, divider}
+
 func Print(keys []pgpkey.PgpKey) {
-	fmt.Printf("%s\n", makeTable(keys))
+	rows := makeTableRows(keys)
+	rowStrings := makeStringsFromRows(rows)
+	for _, rowString := range rowStrings {
+		fmt.Println(rowString)
+	}
 }
 
-func makeTable(keys []pgpkey.PgpKey) string {
-	var table string
-	table += fmt.Sprintf("%s\n", colour.LightBlue(makeHeader(columnWidths(keys))))
-	table += fmt.Sprintf("%s\n", makeHorizontalUnderlines(columnWidths(keys)))
+func makeTableRows(keys []pgpkey.PgpKey) []row {
+	var rows []row
+	rows = append(rows, header)
+	rows = append(rows, placeholderDividerRow)
+	rows = append(rows, makeRowsForKeys(keys)...)
+	return rows
+}
 
+// makeRowsForKeys takes a slice of PgpKeys and returns a slice of rows
+// representing the emails, creation time and warning lines associated for
+// each key.
+// e.g ->   [['jane@example.com', '12 Jan 1998', 'Due for rotation',
+//           ['jane@work.com', '', 'Another warning']]
+// It adds a dividing line between each key
+func makeRowsForKeys(keys []pgpkey.PgpKey) []row {
+	var allRows []row
 	for _, key := range keys {
-		emails := key.Emails(true)
-		if len(emails) == 0 {
-			continue // skip this key
-		}
-		firstEmail := emails[0]
-		table += fmt.Sprintf("%-*s%s", columnWidths(keys)[0], firstEmail, gutter)
-		table += fmt.Sprintf("%-*s", columnWidths(keys)[1], key.PrimaryKey.CreationTime.Format("2 Jan 2006"))
-		table += fmt.Sprintf("\n")
-		for _, email := range emails[1:] {
-			table += fmt.Sprintf("%v\n", email)
+		columns := []column{
+			key.Emails(true),
+			[]string{key.PrimaryKey.CreationTime.Format("2 Jan 2006")},
+			keyWarningLines(key),
 		}
+		keyRows := makeRowsFromColumns(columns)
+		allRows = append(allRows, keyRows...)
+		allRows = append(allRows, placeholderDividerRow)
+	}
+	return allRows
+}
+
+// makeRows takes a slice of columns, and returns a slice of rows.
+// It pads out any of the shorter columns with empty cells.
+// e.g.  Columns  : ["Jim", "Jane", "Fi"], ["1", "2"]
+//       => Rows  : [["Jim", "1"], ["Jane", "2"], ["Fi", ""]]
+func makeRowsFromColumns(columns []column) []row {
```

Comment:
lots of potential for this one blowing up, looking forward to seeing some awesome tests!